### PR TITLE
Remaining type hinting suggestions

### PR DESCRIPTION
**Pull request by sydney-runkle**
Adding remaining suggestions from https://github.com/pydantic/pydantic/pull/7680

Selected Reviewer: @alexmojaki

### CODE REVIEW

**Comment by sydney-runkle** [comment]
Please review

File: pydantic/main.py [hunk 1]
```diff
@@ -32,13 +32,17 @@
 from .json_schema import DEFAULT_REF_TEMPLATE, GenerateJsonSchema, JsonSchemaMode, JsonSchemaValue, model_json_schema
 from .warnings import PydanticDeprecatedSince20
 
+if sys.version_info < (3, 10):
+    TypeAlias = typing_extensions.TypeAlias
```
Comment (alexmojaki) [inline pydantic/main.py:36]: Why not always use this?

Comment (sydney-runkle) [reply pydantic/main.py:36]: We can do that, one moment...

**Comment by alexmojaki** [comment]
Thanks!

**Pull request merged**